projects/plugins/jetpack/_inc/client/components/navigation/index.jsx

```diff
@@ -50,6 +51,17 @@ export class Navigation extends React.Component {
 
 	render() {
 		let navTabs;
+
+		const jetpackPlansPath = getRedirectUrl(
+			this.props.hasConnectedOwner ? 'jetpack-plans' : 'jetpack-nav-site-only-plans',
```

Comment:
Replaced `isLinked` with `hasConnectedOwner` which represents what we really need here.